### PR TITLE
remove unwanted attributes.rails as it cannot be restored

### PR DESCRIPTION
**Pull request by MaximilianMeister**
cc @rhafer

### CODE REVIEW

File: crowbar_framework/lib/crowbar/upgrade.rb [hunk 1]
```diff
@@ -46,7 +46,14 @@ def knife_files
       crowbar_databags_path.children.each do |file|
         file_path = crowbar_databags_path.join(file)
 
-        if file.basename.to_s =~ /^bc-nova_dashboard-(.*)\.json$/
+        case file.basename.to_s
+        when /^bc-crowbar-(.*)\.json$/
+          json = JSON.load(file.read)
+          if json["raw_data"]["attributes"]
+            json["raw_data"]["attributes"].delete("rails")
```
Comment (rhafer) [inline crowbar_framework/lib/crowbar/upgrade.rb]: As ```json``` is of type Chef::DataBagItem  this should be:
```json["attributes"].delete("rails")``` or  ```json.raw_data["attributes"].delete("rails")```
I'd prefer the first.

**Comment by rhafer** [comment]
+1

**Comment by vuntz** [comment]
Isn't that covered by chef/data_bags/crowbar/migrate/crowbar/010_drop_rails.rb?

**Comment by rhafer** [comment]
> Isn't that covered by chef/data_bags/crowbar/migrate/crowbar/010_drop_rails.rb?

@vuntz Yeah, but it can't work. Because the "rails" attribute is outside the "attributes.crowbar" tree. And the migration only get's to see everything inside that hierarchy.

**Comment by vuntz** [comment]
@rhafer then I'm missing the "why is this unwanted?" explanation in the commit message :-)

**Comment by rhafer** [comment]
> then I'm missing the "why is this unwanted?" explanation in the commit message :-)

@MaximilianMeister can you add something like this to the commit message:

The "rails" attributes have been removed from the crowbar proposal. We're deleting them here because it's not possible to delete them using a Schema Migration as they were located outside the "attributes.crowbar" hierarchy in the proposal.

**Comment by jsuchome** [comment]
+1 (and I also vote for more detailed commit message)

**Comment by MaximilianMeister** [comment]
@jsuchome @rhafer @vuntz commit message updated

**Comment by rhafer** [comment]
+1 again

**Comment by vuntz** [comment]
@MaximilianMeister hrm, it seems your lines in the commit messages are a bit too long ;-)

But +1

**Comment by MaximilianMeister** [comment]
@rhafer @vuntz gating is green, can someone merge this please?

**Pull request merged**